### PR TITLE
fix typos present in '@theia/core'

### PR DESCRIPTION
**Pull request by justinang1**
Issue: #6899

#### What it does
* fixes various typos throughout the codebase. 

#### Review checklist
* [x]  as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers
* as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 Breaking changes:
 
+- [core] renamed method `registerComositionEventListeners()` has been renamed to `registerCompositionEventListeners()` [#6912](https://github.com/eclipse-theia/theia/pull/6912)
```
Comment (vince-fugnitto) [inline CHANGELOG.md]: [minor] the wording is a little off, `renamed` is mentioned twice. Something like the following would be more appropriate:

```md
- [core] renamed method `registerComositionEventListeners()` to `registerCompositionListeners()`...
```

**Comment by vince-fugnitto** [comment]
@justinang1 when you get a chance, do you mind fixing the conflicts of the pull-request so it can be merged?

**Comment by justinang1** [comment]
Sure, here's what I plan to do: 
1. Pull from upstream repository
2. Merge + keep both changes to CHANGELOG
3. Push to master

Please let me know if there's any issues to this approach.

**Comment by vince-fugnitto** [comment]
> Please let me know if there's any issues to this approach.

Since you are working off a fork, my approach would be to:
1. sync your fork against the upstream (this repo's) master:
   - https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork
2. rebase your changes against the master
3. fix the conflicts
4. amend your pull-request
5. force push your changes

**Comment by vince-fugnitto** [comment]
Please note that we cannot merge the pull-request until the conflicts have been resolved, and commits are squashed (single commit).

**Comment by justinang1** [comment]
Sorry, I'm a little confused. I merged the fork using github's tool, but now there's two commits. Should I try and resolve this using rebase (Something like git rebase HEAD~2)? Or should I attempt to roll back the repository to my last commit then merge it manually.

**Comment by vince-fugnitto** [comment]
> Sorry, I'm a little confused. I merged the fork using github's tool, but now there's two commits. Should I try and resolve this using rebase (Something like git rebase HEAD~2)? Or should I attempt to roll back the repository to my last commit then merge it manually.

You can do however you feel most comfortable with :)
The only thing to make sure is that commits are squashed (unless there are multiple commits for a specific reason) and that there are no merge conflicts before we are able to successfully merge.

**Pull request closed without merging**